### PR TITLE
Feature/dashboard

### PR DESCRIPTION
**Pull request by b-dabrowski**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,8 +9,8 @@ env:
...
   include:
   - stage: build
-    script: yarn build
+    script: cd backend && yarn build
   - stage: test
-    script: yarn test
+    script: cd backend && yarn test
   - stage: semantic-release
-    script: yarn run semantic-release
+    script: cd backend && yarn run semantic-release
```
Comment (TomaszChmielPagepro) [inline .travis.yml]: I think we should run `semantic-release` for whole repo

File: backend/src/common/definitions/common.ts [hunk 2]
```diff
@@ -2,5 +2,5 @@ export { }
 
 declare global {
   interface IStringTMap<T> { [key: string]: T }
-  type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>
+  // type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>
```
Comment (TomaszChmielPagepro) [inline backend/src/common/definitions/common.ts]: Do we need this or not? 😛

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
```
Comment (TomaszChmielPagepro) [inline build.sh]: Maybe we should add some `echo`'s for each build stage?

File: client-dashboard/src/components/LandingPage.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React, { useState, useEffect } from 'react'
+import landingPageTemplate from '../templates/landingPageTemplate'
+import axios from 'axios'
+
+const LandingPage: React.FC = () => {
+  const [link, setLink] = useState("")
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const { data } = await axios.get('api/installation/slackInstallationLink');
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/components/LandingPage.tsx]: Let's change it to `/api/settings` and add `slackInstallationLink` as a field there

File: .gitignore [hunk 5]
```diff
@@ -62,6 +64,35 @@ typings/
...
+client-dashboard/.pnp.js
+
+# testing
+client-dashboard/coverage
+
+# production
+client-dashboard/build
+
+# misc
+client-dashboard/.DS_Store
```
Comment (klojniewski) [inline .gitignore]: I would ignore all `.DS_Store` files, not just in `client-dashboard` folder.

File: build.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+DIST_FRONTEND_PATH='../backend/dist/frontend/'
+BUILD_FRONEND_PATH='./build/'
+
+echo '---start backend build---'
```
Comment (TomaszChmielPagepro) [inline build.sh]: echo '---BACKEND BUILD STARTING---'

File: build.sh [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+cd 'backend/'
+yarn install
+yarn run build
+echo '---end backend build---'
```
Comment (TomaszChmielPagepro) [inline build.sh]: echo '---BACKEND BUILD SUCCEEDED---'

File: build.sh [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+echo '-----------------------'
+
+echo '---start client build--'
```
Comment (TomaszChmielPagepro) [inline build.sh]: Dash missing at the end

File: client-dashboard/src/layout/SidebarLayout.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    url: dashboardRoutes.giftsManagementPage
+  },
+  {
+    content: pageTitles.title1,
+    iconType: 'apartment',
+    url: dashboardRoutes.settingPage
+  }
+]
+
+class SidebarLayout extends React.Component<RouteComponentProps> {
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/layout/SidebarLayout.tsx]: This could be a function component

File: client-dashboard/src/routers/AppRouter.tsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { Route, Switch, BrowserRouter } from 'react-router-dom'
+import DashboardApp from '../DashboardApp';
+import InfoApp from '../InfoApp';
+import { routes } from '../setup/config';
+
+class AppRouter extends React.Component {
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/routers/AppRouter.tsx]: This could be a function component

File: client-dashboard/src/components/LandingPage.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
+import React, { useState, useEffect } from 'react'
+import landingPageTemplate from '../templates/landingPageTemplate'
+import axios from 'axios'
+
+const LandingPage: React.FC = () => {
+  const [link, setLink] = useState("")
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/components/LandingPage.tsx]: Use String.empty

File: client-dashboard/src/components/SuccessInstallationPage.tsx [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import successInstallationPageTemplate from '../templates/successInstallationTemplate'
+
+const SuccessInstallationPage: React.FC = () => {
+  return (
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/components/SuccessInstallationPage.tsx]: Unecessary `return`

File: client-dashboard/src/components/gifts/GiftPage.tsx [hunk 13]
```diff
@@ -0,0 +1,15 @@
+import { Divider, PageHeader } from 'antd'
+import React from 'react'
+import { pageTitles } from '../../setup/messages'
+
+const GiftPage: React.FC = () => {
+  return (
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/components/gifts/GiftPage.tsx]: Unecessary `return`

File: client-dashboard/src/index.tsx [hunk 14]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './scss/main.css';
+import './setup/extensions'
+import App from './App';
+
+
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/index.tsx]: Unecessary enter

File: client-dashboard/src/layout/FooterLayout.tsx [hunk 15]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { Layout } from 'antd'
+
+const { Footer } = Layout;
+
+const FooterLayout: React.FC = () => {
+  const year = new Date().getFullYear()
+  const message = `Open Kudos ©${year} Created by Pagepro`
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/layout/FooterLayout.tsx]: New line here

File: client-dashboard/src/layout/HeaderLayout.tsx [hunk 16]
```diff
@@ -0,0 +1,11 @@
+import { Layout } from 'antd'
+import React from 'react'
+const { Header } = Layout
+
+const HeaderLayout: React.FC = () => {
+  return (
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/layout/HeaderLayout.tsx]: Unecessary `return`

File: client-dashboard/src/routers/AppRouter.tsx [hunk 17]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Route, Switch, BrowserRouter } from 'react-router-dom'
+import DashboardApp from '../DashboardApp';
+import InfoApp from '../InfoApp';
+import { routes } from '../setup/config';
+
+const AppRouter: React.FC = () => {
+  return (
```
Comment (TomaszChmielPagepro) [inline client-dashboard/src/routers/AppRouter.tsx]: Unecessary `return`

**Pull request merged**